### ISSUE TITLE
Test Output of Tutorial does not align with presented Code

### ISSUE THREAD

**Issue by MyGodItsFull0fStars** [davimiku/json_parser#2]
Hi David,

I am currently working through your JSON parser tutorial and I do enjoy it so far.
As I did not find another way to contact you, I thought I will create an issue (but in reality its a small remark).

In the Writing Tests section, right after adding `#[derive(Debug,...]` to the Token enum within `tokenize.rs`, there is the test output for the following unit test:
```bash
#[test]
fn just_comma() {
    let input = String::from(",");
    let expected = [Token::Comma];

    let actual = tokenize(input);

    assert_eq!(actual, expected); <----- the issue is related to this line!
}
```

Your shown CLI output is:

```bash
assertion `left == right` failed
  left: [Comma]
 right: []
```
The issue with that is, that the `Comma` token should be on the right side (the `expected` variable).
If I execute the test on my machine (because it runs there), the CLI output is as expected:

```bash
assertion `left == right` failed
  left: []
 right: [Comma]
```

Anyway, that is just a minor thing I have found so far, and I really enjoy your explanations of the different topics and I hope to see more of that from you in the future ;)

**Comment by davimiku** [comment]
Thank you! The best compliment I can receive is that people are reading the tutorial and are having fun with it 😄 

Good catch, I think what happened is in my first draft, I was inconsistent across my tests of the order (actual,expected or expected,actual). I cleaned that up later, but didn't update the text of the CLI output.

Thanks! Fixed now in davimiku/blog.davimiku.com@82bfca6e5c40bc3657770b068885c04168f2dbd9

**Comment by MyGodItsFull0fStars** [comment]
Yes I do have fun with the tutorial! You did an amazing job explaining the concepts step by step 😄 

Happens to the best, and it's just a minor thing, but I thought I will mention it if any beginner might stumble over it.

In case I find anything else, shall I post it here or create a separate issue?

**Comment by davimiku** [comment]
Please do! Feel free to post it here, that's fine

**Comment by MyGodItsFull0fStars** [comment]
Hey again David!

I finished your tutorial last week and finally have time for some remarks other than I had fun :D

The remarks are both regarding the `parse.rs` file.

The first remark is for the very first test `parses_null()` (and I guess further on tests as well), that the `let actual = parse_tokens(&input, &mut 0).unwrap()` part will not compile due to the `parse_tokens(...)` signature expecting to take ownership of the `tokens` parameter. 

See below:
```rust
fn parse_tokens(tokens: Vec<Token>, index: &mut usize) -> Result<Value, TokenParseError>
```

To keep all tests as they are presented in your tutorial, I added a borrow to the `tokens` parameter in the `parse_tokens()` signature:
```rust
fn parse_tokens(tokens: &Vec<Token>, index: &mut usize) -> Result<Value, TokenParseError>
```

With this change, the tests module functions will compile as you presented them in your tutorial.
